crates/sui-core/src/authority_active/gossip/mod.rs

```diff
@@ -224,14 +225,19 @@ where
         peer_name: AuthorityName,
         active_authority: &ActiveAuthority<A>,
         start_seq: Option<TxSequenceNumber>,
-    ) -> PeerGossip<A> {
-        PeerGossip {
+    ) -> SuiResult<PeerGossip<A>> {
+        let start_seq = active_authority
+            .state
+            .get_next_expected_sequence(peer_name)?
+            .or(start_seq);
```

Comment:
It doesn't make the None option entirely useless, as that will still be the starting point the first time we connect.

However, I think I've forgotten about how the gossip mesh is supposed to work. We repeatedly replace peers that we follow with new randomly selected peers, right? So if we later return to a peer, this would pick up where we left off, which is not what we want.

@laura-makdah How do you think this should work for gossip?

Reply:
This code is ok with me.

Reply:
@lxfind discussed with Laura on slack, will update you